app/src/main/java/de/triology/cas/ldap/CesAuthenticationEventExecutionPlanConfiguration.java

```diff
@@ -0,0 +1,226 @@
+                getOrder(), authenticator, strategy, groupResolver);
+        handler.setCollectDnAttribute(l.isCollectDnAttribute());
+
+        if (!l.getAdditionalAttributes().isEmpty()) {
+            val additional = CoreAuthenticationUtils.transformPrincipalAttributesListIntoMultiMap(l.getAdditionalAttributes());
+            multiMapAttributes.putAll(additional);
+        }
+        if (StringUtils.isNotBlank(l.getPrincipalDnAttributeName())) {
+            handler.setPrincipalDnAttributeName(l.getPrincipalDnAttributeName());
+        }
+        handler.setAllowMultiplePrincipalAttributeValues(l.isAllowMultiplePrincipalAttributeValues());
+        handler.setAllowMissingPrincipalAttributeValue(l.isAllowMissingPrincipalAttributeValue());
+        handler.setPasswordEncoder(PasswordEncoderUtils.newPasswordEncoder(l.getPasswordEncoder(), applicationContext));
+        handler.setPrincipalNameTransformer(PrincipalNameTransformerUtils.newPrincipalNameTransformer(l.getPrincipalTransformation()));
+
+        if (StringUtils.isNotBlank(l.getCredentialCriteria())) {
+            LOGGER.trace("Ldap authentication for [{}] is filtering credentials by [{}]",
+                    l.getLdapUrl(), l.getCredentialCriteria());
+            handler.setCredentialSelectionPredicate(CoreAuthenticationUtils.newCredentialSelectionPredicate(l.getCredentialCriteria()));
+        }
+
+        if (StringUtils.isBlank(l.getPrincipalAttributeId())) {
+            LOGGER.trace("No principal id attribute is found for LDAP authentication via [{}]", l.getLdapUrl());
+        } else {
+            handler.setPrincipalIdAttribute(l.getPrincipalAttributeId());
+            LOGGER.trace("Using principal id attribute [{}] for LDAP authentication via [{}]", l.getPrincipalAttributeId(), l.getLdapUrl());
+        }
+
+        val passwordPolicy = l.getPasswordPolicy();
+        if (passwordPolicy.isEnabled()) {
+            LOGGER.trace("Password policy is enabled for [{}]. Constructing password policy configuration", l.getLdapUrl());
+            val cfg = createLdapPasswordPolicyConfiguration(passwordPolicy, authenticator, multiMapAttributes);
+            handler.setPasswordPolicyConfiguration(cfg);
+        }
+
+        val attributes = CollectionUtils.wrap(multiMapAttributes);
+        handler.setPrincipalAttributeMap(attributes);
+
+        LOGGER.debug("Initializing LDAP authentication handler for [{}]", l.getLdapUrl());
+        handler.initialize();
+
+        return handler;
+    }
+
+    @Override
+    public void configureAuthenticationExecutionPlan(AuthenticationEventExecutionPlan plan) {
+        plan.registerAuthenticationHandler(cesGroupAwareLdapAuthenticationHandler());
+        LOGGER.trace("Registered {}, registered authentication handlers: {}",
+                CesGroupAwareLdapAuthenticationHandler.class.getSimpleName(), plan.getAuthenticationHandlers());
+    }
+
+    // 1:1 copy from LdapAuthenticationConfiguration
```

Comment:
Why are here comments like this? Which value do they add?

Reply:
I have expanded the commentary.